Topic1/src/com/topic1/MostExpensive.java

```diff
@@ -0,0 +1,31 @@
+package com.topic1;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class MostExpensive implements FindItem {
+
+	HashSet<Item> items;
+
+	public MostExpensive(HashSet<Item> items) {
+		this.items = items;
+
+	}
+
+	public Item find() {
+
+		double expensive = 0;
+		Item eItem = null;
+		Iterator<Item> it = items.iterator();
+		System.out.println(" " + (it.next()).getName()); // without this line
+		// throw exception?
+		while (it.hasNext()) {
+			if (expensive < it.next().getPrice()) {
```

Comment:
[High] Each time that you call to *next* method, you go to next element. If you want to manipulate an element and you need to access several times, you have to store it in an auxiliar variable.